### ISSUE TITLE
Cannot compile POSIX without warnings

### ISSUE THREAD

**Issue by atoomic** [rurban/perl-compiler#335]
env: perl 5.22.0

the same problem can be noticed while generating multiple cp binaries
this warning is a blocker as bincheck fails and it will be displayed at each run

```
> perlcc -r -e 'use POSIX (); print POSIX::sqrt(4)."\n"'
pccgUPV2.c:6252: warning: integer constant is too large for ‘unsigned long’ type
pccgUPV2.c:6253: warning: integer constant is too large for ‘unsigned long’ type
Constant subroutine M_SQRT2 redefined at ./a.out line 4294967295.
2
```

same thing when forcing POSIX to be compiled
```
> perlcc -uPOSIX -r -e 'use POSIX (); print POSIX::sqrt(4)."\n"'
pccm8lMk.c:6252: warning: integer constant is too large for ‘unsigned long’ type
pccm8lMk.c:6253: warning: integer constant is too large for ‘unsigned long’ type
Constant subroutine M_SQRT2 redefined at ./a.out line 4294967295.
2
```

**Comment by atoomic** [comment]
this is probably coming from POSIX.xs
```
ext/POSIX/POSIX.xs:73:#  undef M_SQRT2
ext/POSIX/POSIX.xs:87:#  define M_SQRT2    M_SQRT2q
ext/POSIX/POSIX.xs:104:#    undef M_SQRT2
ext/POSIX/POSIX.xs:144:#  ifndef M_SQRT2
ext/POSIX/POSIX.xs:145:#    define M_SQRT2      FLOAT_C(1.41421356237309504880168872420969808)

**Comment by atoomic** [comment]
```
> grep POSIX a.out.c  | grep boot
    gv_list[46] = gv_fetchpvn_flags("POSIX::bootstrap", 16, GV_ADD, SVt_PV);
    sv_list[1427].sv_u.svu_pv = savepvn("POSIX::bootstrap\000\001", 18);
    sv_list[1459].sv_u.svu_pv = savepvn("boot_POSIX\000\001", 12);
    gv_list[516] = gv_fetchpvn_flags("POSIX::Termios::bootstrap", 25, GV_ADD, SVt_PV);
    gv_list[546] = gv_fetchpvn_flags("POSIX::SigSet::bootstrap", 24, GV_ADD, SVt_PV);
EXTERN_C void boot_POSIX(pTHX_ CV* cv);
        CV* cv = get_cv("POSIX::bootstrap", GV_ADD);
        boot_POSIX(aTHX_ cv);

**Comment by rurban** [comment]
Fixed in master with c1fc5439ab3442b745ae3f438aa57cfe884d97f0

**Comment by atoomic** [comment]
@rurban cool thanks
